### DISCUSSION TITLE
Problem with broadband optimizations

### DISCUSSION THREAD

**Discussion by Guizhixing** [NanoComp/meep#2789] [Q&A] [answered]
Hello everyone! 
I'm new to Meep, and I have a naive question for help.
I am using the adjoins solver to inverse design an electromagnetic device. My objective function is the mean square error (MSE) of the monitored transmission spectrum and the target transmission spectrum. The transmission spectrum contains 20 frequency points, and the design area is 64×64 material grid. After an iteration, I got a gradient array with a dimension of (4096,20). What I wonder is that this gradient should not be (4096,)?

**Comment by smartalecH** [comment, marked as answer]
In meep you have to manually sum the gradients along the frequency axis. See a previous discussion on the topic in #2773.
